applications/nrf5340_audio/src/bluetooth/bt_stream/unicast/unicast_client.c

```diff
@@ -1436,87 +1362,90 @@ int unicast_client_discover(struct bt_conn *conn, enum unicast_discover_dir dir)
-				LOG_WRN("Failed to enable stream %d: %d", i, ret);
-				any_errors = true;
-			}
-
-		} else if (dir == BT_AUDIO_DIR_SOURCE &&
-			   le_audio_ep_state_check(headsets[i].source_stream.ep,
-						   BT_BAP_EP_STATE_QOS_CONFIGURED)) {
+			cap_stream_params[param.count].member.member = headsets[i].headset_conn;
+			cap_stream_params[param.count].stream = &headsets[i].sink_stream;
+			cap_stream_params[param.count].ep = headsets[i].sink_ep;
+			cap_stream_params[param.count].codec_cfg = &lc3_preset_sink.codec_cfg;
+			param.count++;
+		}
+		if (le_audio_ep_state_check(headsets[i].source_ep, BT_BAP_EP_STATE_IDLE)) {
 			/* Start all streams in the configured state */
-			ret = bt_bap_stream_enable(&headsets[i].source_stream,
-						   lc3_preset_source.codec_cfg.meta,
-						   lc3_preset_source.codec_cfg.meta_len);
-
-			if (ret) {
-				LOG_WRN("Failed to enable stream %d: %d", i, ret);
-				any_errors = true;
-			}
+			cap_stream_params[param.count].member.member = headsets[i].headset_conn;
+			cap_stream_params[param.count].stream = &headsets[i].source_stream;
+			cap_stream_params[param.count].ep = headsets[i].source_ep;
+			cap_stream_params[param.count].codec_cfg = &lc3_preset_source.codec_cfg;
+			param.count++;
 		}
 	}
 
-	if (any_errors) {
+	if (param.count > 0) {
+		ret = bt_cap_initiator_unicast_audio_start(&param);
+		if (ret) {
+			LOG_ERR("Failed to start unicast audio: %d", ret);
+			return ret;
+		}
+
+		playing_state = true;
+	} else {
 		return -EIO;
 	}
 
-	playing_state = true;
-
 	return 0;
 }
 
-int unicast_client_stop(enum bt_audio_dir dir)
+int unicast_client_stop(void)
 {
 	int ret;
-	bool any_errors = false;
+	struct bt_cap_stream *streams[CONFIG_BT_BAP_UNICAST_CLIENT_ASE_SRC_COUNT +
+				      CONFIG_BT_BAP_UNICAST_CLIENT_ASE_SNK_COUNT];
+	static struct bt_cap_unicast_audio_stop_param param;
+
+	param.streams = streams;
+	param.count = 0;
+	param.type = BT_CAP_SET_TYPE_AD_HOC;
 
 	le_audio_event_publish(LE_AUDIO_EVT_NOT_STREAMING, NULL, 0);
 
 	for (int i = 0; i < ARRAY_SIZE(headsets); i++) {
-		if (dir == BT_AUDIO_DIR_SINK &&
-		    le_audio_ep_state_check(headsets[i].sink_stream.ep,
-					    BT_BAP_EP_STATE_STREAMING)) {
-			/* Stop all streams currently in a streaming state */
-			ret = bt_bap_stream_disable(&headsets[i].sink_stream);
-
-			if (ret) {
-				LOG_WRN("Failed to disable stream %d: %d", i, ret);
-				any_errors = true;
-			}
-
-		} else if (dir == BT_AUDIO_DIR_SOURCE &&
-			   le_audio_ep_state_check(headsets[i].source_stream.ep,
-						   BT_BAP_EP_STATE_STREAMING)) {
-			/* Stop all streams currently in a streaming state */
-			ret = bt_bap_stream_disable(&headsets[i].source_stream);
-
-			if (ret) {
-				LOG_WRN("Failed to disable stream %d: %d", i, ret);
-				any_errors = true;
-			}
+		if (le_audio_ep_state_check(headsets[i].sink_ep, BT_BAP_EP_STATE_STREAMING)) {
+			/* Stop all sink streams currently in a streaming state */
+			streams[param.count] = &headsets[i].sink_stream;
+			param.count++;
```

Comment:
Discuss: Usually this `param.count` is used as a boolean. Can we change to bool, or will this have negative effects elsewhere?

Reply:
It is not a boolean and is not used as one either. It is the number of pointers given to start/stop